zonemaster/zonemaster-engine
# Update implementation of DNSSEC03

## Purpose

This PR proposes an update to the implementation of DNSSEC03 following an update to the specification (https://github.com/zonemaster/zonemaster/pull/1189).

Note that Prefix Suffix List check (step 13.2.2.1) is not yet implemented, as this service is not yet provided by Zonemaster.

Requires a fix to Zonemaster::LDNS::RR::NSEC3 to work: https://github.com/zonemaster/zonemaster-ldns/pull/177

## Context

Test Case specification: https://github.com/zonemaster/zonemaster/pull/1189
Test Zones specification: https://github.com/zonemaster/zonemaster/pull/1218

Requires https://github.com/zonemaster/zonemaster-ldns/pull/177

## Changes

`lib/Zonemaster/Engine/Test/DNSSEC.pm`

- Update implementation
- Update message tags and message ids

`share/profile.json` and `share/profile.yaml`

- Update message tags

`t/Test-dnssec03.t` and `t/Test-dnssec03.data`

- Update unitary tests implementation and data

## How to test this PR

Unit tests are updated and should pass.

**Manual testing:**
```
$ git log -1 --oneline
25036e6 (HEAD -> update-dnssec03, origin/update-dnssec03) Update implementation of DNSSEC03

$ zonemaster-cli --test dnssec/dnssec03 --show-testcase --level INFO zonemaster.net
Seconds Level    Testcase       Message
======= ======== ============== =======
   0.01 INFO     UNSPECIFIED    Using version v4.7.3 of the Zonemaster engine.
   5.79 INFO     DNSSEC03       The zone does not use NSEC3. Testing for NSEC3 has been skipped. Fetched from name servers "ns2.nic.fr/192.93.0.4;ns2.nic.fr/2001:660:3005:1::1:2;nsa.dnsnode.net/194.58.192.46;nsa.dnsnode.net/2a01:3f1:46::53;nsp.dnsnode.net/194.58.198.32;nsp.dnsnode.net/2a01:3f1:3032::53;nsu.dnsnode.net/185.42.137.98;nsu.dnsnode.net/2a01:3f0:400::32".

$ zonemaster-cli --test dnssec/dnssec03 --show-testcase --level INFO afnic.fr
Seconds Level    Testcase       Message
======= ======== ============== =======
   0.00 INFO     UNSPECIFIED    Using version v4.7.3 of the Zonemaster engine.
   3.66 INFO     DNSSEC03       The following servers respond with a legal hash algorithm in NSEC3. Fetched from name servers "g.ext.nic.fr/194.0.36.1;g.ext.nic.fr/2001:678:4c::1;ns1.nic.fr/192.134.4.1;ns1.nic.fr/2001:67c:2218:2::4:1;ns2.nic.fr/192.93.0.4;ns2.nic.fr/2001:660:3005:1::1:2;ns3.nic.fr/192.134.0.49;ns3.nic.fr/2001:660:3006:1::1:1".
   3.66 NOTICE   DNSSEC03       The following servers respond with NSEC3 opt-out enabled. The recommended practice is to disable opt-out. Fetched from name servers "g.ext.nic.fr/194.0.36.1;g.ext.nic.fr/2001:678:4c::1;ns1.nic.fr/192.134.4.1;ns1.nic.fr/2001:67c:2218:2::4:1;ns2.nic.fr/192.93.0.4;ns2.nic.fr/2001:660:3005:1::1:2;ns3.nic.fr/192.134.0.49;ns3.nic.fr/2001:660:3006:1::1:1".
   3.66 INFO     DNSSEC03       The following servers respond with NSEC3 iteration value set to zero (as recommended). Fetched from name servers "g.ext.nic.fr/194.0.36.1;g.ext.nic.fr/2001:678:4c::1;ns1.nic.fr/192.134.4.1;ns1.nic.fr/2001:67c:2218:2::4:1;ns2.nic.fr/192.93.0.4;ns2.nic.fr/2001:660:3005:1::1:2;ns3.nic.fr/192.134.0.49;ns3.nic.fr/2001:660:3006:1::1:1".
   3.66 INFO     DNSSEC03       The following servers respond with a legal empty salt in NSEC3. Fetched from name servers "g.ext.nic.fr/194.0.36.1;g.ext.nic.fr/2001:678:4c::1;ns1.nic.fr/192.134.4.1;ns1.nic.fr/2001:67c:2218:2::4:1;ns2.nic.fr/192.93.0.4;ns2.nic.fr/2001:660:3005:1::1:2;ns3.nic.fr/192.134.0.49;ns3.nic.fr/2001:660:3006:1::1:1".
```
